src/test/java/ui/UITests.java

```diff
@@ -0,0 +1,111 @@
+package ui;
+
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.Point;
+import org.openqa.selenium.WebElement;
+import ui.helpers.ActionsHelper;
+import ui.pages.DemoApiTestsOneLaunchPage;
+import ui.pages.DemoDashboardPage;
+import ui.pages.LaunchPage;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static ui.logger.UILogger.LOGGER;
+
+
+public class UITests extends BaseUITest {
+    private LaunchPage launchPage = new LaunchPage();
+    private DemoApiTestsOneLaunchPage demoApiTestsOneLaunchPage = new DemoApiTestsOneLaunchPage();
+    private DemoDashboardPage demoDashboardPage = new DemoDashboardPage();
+
+    @Test
+    void scrollToElementWithJS() {
+        assertTrue(launchPage.openLaunchPage().scrollToDefinedElement());
+    }
+
+    @Test
+    void clickOnElementWithJS() {
+        launchPage.openLaunchPage().scrollToDefinedElement();
+        launchPage.clickOnDefinedElement();
+        assertTrue(demoApiTestsOneLaunchPage.validateDemoApiTestsOneLaunchPage());
+    }
+
+    @Test
+    void resizeWidget() {
+        LOGGER.info("Preparing test data ...");
+        widgetId1 = apiSteps.createAndAddWidget(widget);
```

Comment:
Why do you create WidgetController object in BaseUITest, and send it from UI tests to Service ApiSteps class?

Reply:
rewrote ApiSteps class

Reply:
How about to move further and analyze your code for similar situations?
E.g., you login in your ```@BeforeAll``` ```setupDriver``` method.

Reply:
done